int32 entityid
string[] actions
string[] exceptions
string[] temps